File: examples/plugins/provider-python/provider.py
import gestalt

GREETING = "Hello"


class GreetInput(gestalt.Model):
    name: str = gestalt.field(description="Name to greet", default="World")


class GreetOutput(gestalt.Model):
    message: str


def configure(_name: str, config: dict[str, object]) -> None:
    global GREETING
    GREETING = str(config.get("greeting", "Hello"))


@gestalt.operation(
    method="GET",
    description="Return a greeting message",
    read_only=True,
)
def greet(input: GreetInput, _req: gestalt.Request) -> GreetOutput:
    return GreetOutput(message=f"{GREETING}, {input.name}!")

File: sdk/typescript/tests/fixtures/basic-provider-default-export/provider.ts
import { defineApp, ok, operation, s } from "../../../src/index.ts";

const provider = defineApp({
  operations: [
    operation({
      id: "hello",
      method: "POST",
      input: s.object({
        name: s.string({
          default: "World",
        }),
      }),
      output: s.object({
        message: s.string(),
      }),
      handler(input) {
        return ok({
          message: `Hello, ${input.name}.`,
        });
      },
    }),
  ],
});

export default provider;
